### PR TITLE
button: Convert to vanilla JS

### PR DESCRIPTION
**Pull request by XhmikosR**
Tasks:

- [x] Convert buttons.js
- [ ] Fix tests

### CODE REVIEW

**Comment by Johann-S** [comment]
it seems your tests failed because phantomsJS don't know `closest` function and the polyfill is in my PR, before finishing this PR we should finish the Alert plugin and the Button plugin will be finish quickly after that 👍

**Comment by Johann-S** [comment]
`v4-without-jquery` is up to date with `v4-dev` so you can rebase 👍

File: js/src/dom/selectorEngine.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ const SelectorEngine = {
     return fnMatches.call(element, selector)
   },
 
-  find(selector) {
+  find(element = document, selector) {
```
Comment (Johann-S) [inline js/src/dom/selectorEngine.js]: Nice change 👍 I did the same but without the default value and your change is much better 😄

**Comment by Johann-S** [comment]
Can you update our button visual test ?

File: js/src/button.js [hunk 2]
```diff
@@ -148,39 +151,56 @@ const Button = (($) => {
...
+      }
+      data.toggle()
     })
-    .on(Event.FOCUS_BLUR_DATA_API, Selector.DATA_TOGGLE_CARROT, (event) => {
-      const button = $(event.target).closest(Selector.BUTTON)[0]
-      $(button).toggleClass(ClassName.FOCUS, /^focus(in)?$/.test(event.type))
+
+  EventHandler.on(document,
+    Event.FOCUS_BLUR_DATA_API, Selector.DATA_TOGGLE_CARROT, (event) => {
+      const button = SelectorEngine.closest(event.target, Selector.BUTTON)[0]
```
Comment (XhmikosR) [inline js/src/button.js]: @Johann-S: this `[0]` indeed doesn't seem right.

**Pull request merged**